agent/config/parse_windows.go

```diff
@@ -16,23 +16,108 @@
 package config
 
 import (
+	"fmt"
 	"os"
 	"strings"
 	"syscall"
 	"unsafe"
 
 	"golang.org/x/sys/windows/registry"
 
+	"github.com/aws/amazon-ecs-agent/agent/statemanager/dependencies"
 	"github.com/aws/amazon-ecs-agent/agent/utils"
 	"github.com/cihub/seelog"
 )
 
-const (
-	// envSkipDomainJoinCheck is an environment setting that can be used to skip
-	// domain join check validation. This is useful for integration and
-	// functional-tests but should not be set for any non-test use-case.
-	envSkipDomainJoinCheck = "ZZZ_SKIP_DOMAIN_JOIN_CHECK_NOT_SUPPORTED_IN_PRODUCTION"
-)
+var winRegistry dependencies.WindowsRegistry
+
+func init() {
+	winRegistry = dependencies.StdRegistry{}
+}
+
+func setWinRegistry(mockRegistry dependencies.WindowsRegistry) {
+	winRegistry = mockRegistry
+}
+
+func getInstallationType(installationType string) (string, error) {
+	if installationTypeFull == installationType {
+		return "FULL", nil
+	} else if installationTypeCore == installationType {
+		return "CORE", nil
+	} else {
+		err := seelog.Errorf("unsupported Installation type:%s", installationType)
+		return "", err
+	}
+}
+
+func getReleaseIdForSACReleases(productName string) (string, error) {
+	if strings.HasPrefix(productName, windowsServer2019) {
+		return "2019", nil
+	} else if strings.HasPrefix(productName, windowsServer2016) {
+		return "2016", nil
+	}
+	err := seelog.Errorf("unsupported productName:%s for Windows SAC Release", productName)
+	return "", err
+}
+
+func getReleaseIdForLTSCReleases(releaseId string) (string, error) {
+	if releaseId2004SAC == releaseId {
+		return releaseId2004SAC, nil
+	} else if releaseId1909SAC == releaseId {
+		return releaseId1909SAC, nil
+	}
+	err := seelog.Errorf("unsupported ReleaseId:%s for Windows LTSC Release", releaseId)
+	return "", err
+}
+
+// GetOperatingSystemFamily() reads the NT current version from windows registry and constructs operating system family string
+// In case of any exception this method just returns "windows" as operating system type.
+func GetOperatingSystemFamily() string {
+	key, err := winRegistry.OpenKey(ecsWinRegistryRootKey, ecsWinRegistryRootPath, registry.QUERY_VALUE)
+	if err != nil {
+		seelog.Errorf("unable to open Windows registry key to determine Windows version: %v", err)
+		return unsupportedWindowsOS
+	}
+	defer key.Close()
+
+	productName, _, err := key.GetStringValue("ProductName")
+	if err != nil {
+		seelog.Errorf("unable to read registry key, ProductName: %v", err)
+		return unsupportedWindowsOS
+	}
+	installationType, _, err := key.GetStringValue("InstallationType")
```

Comment:
nit: these could be changed to: 
```
if installationType, _, err := key.GetStringValue("InstallationType"); err != nil {

}
```

Reply:
Sure, It looks the current way more readable to me , will keep the same if you don't mind. This looks a little hard to understand but seems advanced  go style :)